### PR TITLE
Fixed Proposal Details Page, Create CDP Button and CDP List

### LINKED ISSUE #438: [KAVA] Proposal Details Page & Create CDP throws errors
**Describe the bug**
1) The proposal details page throws error when clicking on proposal Title.
2)`Create CDP` button throws error when pressed.

**To Reproduce**
1. Go to 'www.kava.bigdipper.live'
2. Click on 'Proposals'
3. Click on any Proposal Title

**Screenshots**
Screenshot 2020-12-16 at 1 32 33 PM

### PR DESCRIPTION
**Pull request by MonikaCat**
## Description

Fixes #438 

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Run `meteor npm run lint`
- [x] Added an entry to the `CHANGELOG.md` file.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Review by kwunyeung — CHANGES_REQUESTED** [review]
It shows NaN and undefined when there is no available `Hard` in the account.

image

I have claimed some vesting Hard in our account and the detail is not being shown correctly.

image

This is the account address `kava14kn0kk33szpwus9nh8n87fjel8djx0y08wynpx`. Please see if the structure with the vesting token breaks the page.

**Comment by kwunyeung** [comment]
[CDP list display and Ledger button fix](https://trello.com/c/kvIa3fBP/6-cdp-list-display-and-ledger-button-fix)

**Review by kwunyeung — COMMENTED** [review]
Does this means there are different CDPs from different denoms? Should we add a selector for user to choose which collateral they want to explore?

**Comment by kwunyeung** [comment]
I have just done a query `kvcli q cdp cdps --collateral-type hard-a` and there are a lot of CDPs.

**Comment by MonikaCat** [comment]
> I have just done a query `kvcli q cdp cdps --collateral-type hard-a` and there are a lot of CDPs.

Ok I see! Looks like the `cdp/cdps` query tracks only CDPs created with BNB. I've changed that and added Tab selection for each collateral type to allow users explore the collateral they're interested in.

File: imports/ui/accounts/Account.jsx [hunk 1]
```diff
@@ -695,9 +739,14 @@ export default class AccountDetails extends Component {
...
                                     </Row>
                                     <Row>
-                                        <Col xs={12} className="value-3 text-right">{this.findCoin(this.state.total, coin3)}</Col>
-                                        <Col xs={12} className="dollar-value-3 text-right text-secondary">~{numbro((this.findValue(this.state.total, coin3)) * this.state.usdxPrice).format("$0,0.0000a")} ({numbro(this.state.usdxPrice).format("$0,0.00")}/{Meteor.settings.public.coins[5].displayName})</Col>
+                                        <Col xs={12} className="value-3 text-right">{this.findCoin(this.state.total, coin4)}</Col>
+                                        <Col xs={12} className="dollar-value-3 text-right text-secondary">~{numbro((this.findValue(this.state.total, coin4)) * this.state.hardPrice).format("$0,0.0000a")} ({numbro(this.state.hardPrice).format("$0,0.00")}/{Meteor.settings.public.coins[8].displayName})</Col>
```
Comment (kwunyeung) [inline imports/ui/accounts/Account.jsx:743]: I can't test the account operations as I don't know how your `settings.json` looks like right now. Please provide information on the coin changes in the setting as it's all hard coded right now.

File: imports/ui/cdp/CDPs.jsx [hunk 2]
```diff
@@ -33,9 +60,89 @@ export default class CDPs extends Component{
...
+                    </Switch>
+                </TabPane>
+                <TabPane tabId="cdp-bnb">
+                    <Switch>
+                        <Route exact path="/cdps" component={this.CDPList} />
+                    </Switch>
+                </TabPane>
+                <TabPane tabId="cdp-hard">
+                    <Switch>
+                        <Route exact path="/cdps" component={this.CDPList} />
```
Comment (kwunyeung) [inline imports/ui/cdp/CDPs.jsx:127]: What's the usage of creating multiple routes with the same route here?

File: imports/ui/cdp/CDPs.jsx [hunk 3]
```diff
@@ -5,22 +5,49 @@ import List from '/imports/ui/cdp/List.jsx';
...
+        this.state = {
+            cdpActiveTab: 'cdp-kava',
+            collateralType: 'ukava-a'
+        }
+    }
+
+
+    toggleTab = (tab, newCollateralType) => {
+        if (this.state.cdpActiveTab !== tab) {
+            this.setState({
```
Comment (kwunyeung) [inline imports/ui/cdp/CDPs.jsx:30]: If the tabs are all the same, how about not switching tab here and only use one `TabPane` then use pass the correct prop to the `CDPList` component to get a new lis? I see you have `props` in `CDPList` but not passing any props into to it.

**Review by kwunyeung — CHANGES_REQUESTED** [review]

File: imports/ui/cdp/CDPs.jsx (hunk 2 above)
Comment (MonikaCat) [reply imports/ui/cdp/CDPs.jsx]: Ohh! Fixed that!

File: imports/ui/accounts/Account.jsx (hunk 1 above)
Comment (MonikaCat) [reply imports/ui/accounts/Account.jsx:743]: Ok! I will send you my settings.json file now

File: imports/ui/cdp/CDPs.jsx (hunk 3 above)
Comment (MonikaCat) [reply imports/ui/cdp/CDPs.jsx:30]: Ok updated!

**Review by kwunyeung — COMMENTED** [review]
The code LGTM. I'm going to test now.

**Comment by kwunyeung** [comment]
2 issues.

1. I think the CDP calculations with HARD is wrong. The usd value is not correct.
image

2. When switching different denom in CDP, it's better to clear the list and show a loading indicator then wait for new data to feed in. Currently, when you switch the denom in different tabs, it keeps waiting but the existing data keeps there and it's strange. For example, when you are reading the CDP list of Kava and then you switch BNB, the list wouldn't be refreshed until the new data have arrived. In that case the tab has been switched to BNB but the CDP list is still showing the CDP of Kava.

**Review by kwunyeung — APPROVED** [review]

**Pull request merged**